File: src/Cxx/Utilities/PiecewiseFunction.cxx
#include <vtkSmartPointer.h>
#include <vtkPiecewiseFunction.h>

int main(int, char *[])
{
  vtkSmartPointer<vtkPiecewiseFunction> piecewiseFunction = 
      vtkSmartPointer<vtkPiecewiseFunction>::New();
  piecewiseFunction->AddPoint(0.0, 0.0);
  piecewiseFunction->AddPoint(1.0, 2.0);
  
  double test = piecewiseFunction->GetValue(0.25);

  std::cout << "test: " << test << " (should be 0.5)" << std::endl;

  return EXIT_SUCCESS;
}

File: src/Java/Utilities/PiecewiseFunction.java
import vtk.vtkNativeLibrary;
import vtk.vtkPiecewiseFunction;

public class PiecewiseFunction 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

  public static void main(String args[]) 
  {
   vtkPiecewiseFunction piecewiseFunction = new vtkPiecewiseFunction();
   piecewiseFunction.AddPoint(0.0, 0.0);
   piecewiseFunction.AddPoint(1.0, 2.0);
   
   double test = piecewiseFunction.GetValue(0.25);
   
   System.out.println( "Test: "  + " " + test); //(should be 0.5)
  }
}
